component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java

```diff
@@ -1688,4 +1706,92 @@ public static RelationshipManager getRelationshipManager() {
     }
     return relationshipManager;
   }
+
+  public static SiteRestEntity toSiteRestEntity(PortalConfig site, HttpServletRequest request, SiteFilter siteFilter) {
+    if (site == null) {
+      return null;
+    }
+    SiteType siteType = SiteType.valueOf(site.getType().toUpperCase());
+    String displayName = site.getLabel();
+    org.exoplatform.services.security.Identity userIdentity = ConversationState.getCurrent().getIdentity();
+    if (SiteType.GROUP.equals(siteType)) {
+      try {
+        Group siteGroup = getOrganizationService().getGroupHandler().findGroupById(site.getName());
+        if (siteGroup == null || !userIdentity.isMemberOf(siteGroup.getId())) {
+          return null;
+        } else if (org.apache.commons.lang3.StringUtils.isBlank(displayName)) {
+          displayName = siteGroup.getLabel();
+        }
+      } catch (Exception e) {
+        LOG.error("Error while retrieving group with name ", site.getName(), e);
+      }
+    }
+    List<Map<String, Object>> accessPermissions = computePermissions(site.getAccessPermissions());
+    Map<String, Object> editPermission = computePermission(site.getEditPermission());
+
+    UserNode rootNode = null;
+    if (siteFilter.isExpandNavigations()) {
+      String currentUser = userIdentity.getUserId();
+      try {
+        HttpUserPortalContext userPortalContext = new HttpUserPortalContext(request);
+        UserPortalConfig userPortalCfg = getUserPortalConfigService().getUserPortalConfig(site.getName(), currentUser, userPortalContext);
+        UserPortal userPortal = userPortalCfg.getUserPortal();
+        UserNavigation navigation = userPortal.getNavigation(new SiteKey(siteType.getName(), site.getName()));
+        rootNode = userPortal.getNode(navigation, Scope.ALL, UserNodeFilterConfig.builder().build(), null);
+      } catch (Exception e) {
+        LOG.error("Error while getting site {} navigations for user {}", site.getName(), currentUser, e);
+      }
+    }
+    return new SiteRestEntity(siteType,
+            site.getName(),
+            !org.apache.commons.lang.StringUtils.isBlank(displayName) ? displayName : site.getName(),
```

Comment:
add this import org.apache.commons.lang.StringUtils explicitly with the list of imports